packages/core/src/bodl/bodl-emitter-service.ts

```diff
@@ -212,10 +212,16 @@ export default class BodlEmitterService implements BodlService {
                 brand_name: item.brand,
                 currency: currencyCode,
                 category_names: item.categoryNames || [],
+                retail_price: item.retailPrice,
             };
         });
 
-        return [...customItems, ...physicalAndDigitalItems, ...giftCertificateItems];
+        return [...customItems, ...physicalAndDigitalItems, ...giftCertificateItems].map(
+            (item) => ({
+                ...item,
+                product_id: String(item.product_id),
```

Comment:
@bc-peng,
Yes, it's a requirement for BODL. Generally it's because of gift certificates. They don't have product_id, so in BODL context, gift_certificate_id becomes product_id. This field has UUID format, which is string. So it has been decided, that product_id will be always string for BODL events